addons/stock/res_config.py

```diff
@@ -27,23 +25,22 @@ def create_transit_location(self, cr, uid, company_id, company_name, context=Non
             'company_id': company_id,
             'location_id': parent_loc,
         }
-        location_id = self.pool.get('stock.location').create(cr, uid, location_vals, context=context)
-        self.write(cr, uid, [company_id], {'internal_transit_location_id': location_id}, context=context)
+        location_id = self.env['stock.location'].create(location_vals)
```

Comment:
**create** method return the recordset so don't attech postfix **_id** its shold be like : 
```
location = self.env['stock.location'].create(location_vals)
```